pkg/kubelet/kubelet.go

```diff
@@ -2716,8 +2716,8 @@ func (kl *Kubelet) HandlePodReconcile(pods []*v1.Pod) {
 		// been evicted, so if this is about minimizing the time to react to an eviction we
 		// can do better. If it's about preserving pod status info we can also do better.
 		if eviction.PodIsEvicted(pod.Status) {
-			if podStatus, err := kl.podCache.Get(pod.UID); err == nil {
-				kl.containerDeletor.deleteContainersInPod("", podStatus, true)
+			if _, err := kl.podCache.Get(pod.UID); err == nil {
```

Comment:
I think this might actually be wrong - the comment above was last left by me because this block is indeed questionable.  I think the right action is to remove this entire if check, and find out whether evicted pods are correctly terminated with the same latency.

We should in this PR:

1. Remove this if block completely
2. Verify that with this if block removed, an evicted pod has no unnecessary delay to removal
3. If it does have delay, find out why and deliver a more targeted fix

The end outcome is that *no one* except eviction and pod worker should be deleting containers.  If we can completely remove this block safely, I believe your problem is fixed.  However, we need to do a bit more testing to ensure that is the right outcome.